src/dsp_tools/utils/uri_util.py

```diff
@@ -12,3 +12,55 @@ def is_uri(s: str) -> bool:
     fragment = r"(?<fragment>#[\w_.\-~:/]*)"
     m = regex.match(rf"{scheme}:(//{host}{port}?){path}*{query}*{fragment}?", s, flags=regex.UNICODE)
     return m is not None
+
+
+def is_iiif_uri(uri: str) -> bool:
+    """
+    Checks if the given URL is a valid IIIF URL.
+    It should support all versions of IIIF servers,
+    but was constructed with the syntax described in: https://iiif.io/api/image/3.0/#2-uri-syntax
+    IIIF is open to modifications which are not described in the official documentation.
+    https://iiif.io/api/annex/notes/design_principles/#define-success-not-failure
+    This may need to be updated in case of a valid version which is not in the regex.
+
+    Args:
+        uri: The URI to be checked.
+
+    Returns:
+        True if the URI is a valid IIIF URI, False otherwise.
+    """
+    # {scheme}://{server}{/prefix}/{identifier}/{region}/{size}/{rotation}/{quality}.{format}
+    split_uri = uri.lower().split("/")
+    if len(split_uri) < 8:
+        return False
+    # %5E is the URL encoded version of ^ ->
+    # because we do change the uri to lower case we need to change that in the regex
+    # (\d+(\.\d+)?) -> number, can be integer or float
+    # everything needs to encapsulated in a group
+    # otherwise the ^ and $ only apply respectively to the first and last listed in the alternative options
```

Comment:
I don't understand this sentence

Reply:
I was able to remove all these brackets (see screenshot), and the tests still pass:

![Screenshot 2024-05-17 at 10 07 24](https://github.com/dasch-swiss/dsp-tools/assets/39048939/e966e3c2-68c7-4a05-a196-ddbb0c9736c4)

Reply:
let me know if I shall commit that, then you don't have to do it manually

Reply:
Added more tests, now try again.

Reply:
still passing

Reply:
Not for me
![Uploading Screenshot 2024-05-17 at 10.17.50.png…]()

Reply:
Ok, there were a couple too many, but not all of them. The outer most still have to stay for the tests to pass. Except for the last

Reply:
Strange, because we run the same tests. I was so free and committed some more bracket removals. In a minute, we'll see what GitHub CI says. We can then still revert my commit if there is a problem with it.

Reply:
The tests pass, so my bracket removal was valid. But feel free to revert my commit if you prefer to have some more brackets (e.g. if you think it's more readable to have some more brackets)